zstd/decoder.go

```diff
@@ -289,9 +289,15 @@ func (d *Decoder) DecodeAll(input, dst []byte) ([]byte, error) {
 	}
 
 	// Allocation here:
-	br := byteBuf(input)
+	br := frame.bBuf
```

Comment:
If `bBuf` would be just `[]byte`, then the code on lines 292-297 could be simplified to:
```go
br := &frame.bBuf
*br = input
```
And it still shouldn't require memory allocations

Reply:
And it would be better to put `frame.bBuf = nil` before exiting from the function, so GC could collect byte slice pointed by `input`